vvvv45/addonpack/src/nodes/plugins/Animation/DelayFramebased/DelayFramebasedNode.cs

```diff
@@ -0,0 +1,117 @@
+#region usings
+using System;
+using System.ComponentModel.Composition;
+
+using System.Collections.Generic;
+using System.Collections;
+
+using VVVV.PluginInterfaces.V1;
+using VVVV.PluginInterfaces.V2;
+using VVVV.Utils.VColor;
+using VVVV.Utils.VMath;
+
+using VVVV.Core.Logging;
+#endregion usings
+
+namespace VVVV.Nodes
+{
+	#region PluginInfo
+	[PluginInfo(Name = "Delay", Category = "Animation", Version ="Framebased", Help = "Framebased delay with adjustable frame count")]
+	#endregion PluginInfo
+	public class DelayFramebased : IPluginEvaluate
+	{
+		#region fields & pins
+		[Input("Input", DefaultValue = 1.0)]
+		public ISpread<double> FInput;
+		
+		[Input("Skip")]
+		public ISpread<bool> FSkip;
+		
+		[Input("Reset")]
+		public ISpread<bool> FReset;
+
+		[Input("Delay", DefaultValue = 1.0, MinValue = 0)]
+		public ISpread<int> FDelayFrames;
+			
+		[Output("Output")]
+		public ISpread<double> FOutput;
+		
+		[Import()]
+		public ILogger FLogger;
+		#endregion fields & pins
+
+		List<Queue<double>> listOfQueue = new List<Queue<double>>();
+		
+		public void Evaluate(int SpreadMax)
+		{
+			FOutput.SliceCount = SpreadMax;
+			
+			
+			if(SpreadMax != listOfQueue.Count){
+				
+				int difference = SpreadMax - listOfQueue.Count;
+					
+				if(difference < 0){
```

Comment:
while i'm at it: it is good practice to put a space between if and the opening bracket. same with "for ("